python/sidecar/search.py

```diff
@@ -0,0 +1,256 @@
+                "both",
+                "each",
+                "few",
+                "more",
+                "most",
+                "other",
+                "some",
+                "such",
+                "no",
+                "nor",
+                "not",
+                "only",
+                "own",
+                "same",
+                "so",
+                "than",
+                "too",
+                "very",
+                "s",
+                "t",
+                "can",
+                "will",
+                "just",
+                "don",
+                "don't",
+                "should",
+                "should've",
+                "now",
+                "d",
+                "ll",
+                "m",
+                "o",
+                "re",
+                "ve",
+                "y",
+                "ain",
+                "aren",
+                "aren't",
+                "couldn",
+                "couldn't",
+                "didn",
+                "didn't",
+                "doesn",
+                "doesn't",
+                "hadn",
+                "hadn't",
+                "hasn",
+                "hasn't",
+                "haven",
+                "haven't",
+                "isn",
+                "isn't",
+                "ma",
+                "mightn",
+                "mightn't",
+                "mustn",
+                "mustn't",
+                "needn",
+                "needn't",
+                "shan",
+                "shan't",
+                "shouldn",
+                "shouldn't",
+                "wasn",
+                "wasn't",
+                "weren",
+                "weren't",
+                "won",
+                "won't",
+                "wouldn",
+                "wouldn't",
+            ]
+        )
+
+    # Method to preprocess the query
+    def preprocess_query(self, query):
+        # convert query to lowercase and remove stopwords
+        query = query.lower()
+        query = "+".join([word for word in query.split() if word not in self.stopwords])
+        return query
+
+    # Method to perform the search
+    # Other search fields can be found here:
+    # https://github.com/danielnsilva/semanticscholar/blob/da433a1f45e0c17bbd58300de0602c969602675a/semanticscholar/Paper.py#L93
+    def search(
+        self,
+        query,
+        limit=10,
+        search_fields=[
+            "title",
+            "abstract",
+            "venue",
+            "year",
+            "corpusId",
+            "citationCount",
+            "openAccessPdf",
+        ],
+    ):
+        # Preprocess the query
+        query = self.preprocess_query(query)
+        results = self.s2.search_paper(query, limit=limit, fields=search_fields)
+        results_list = []
+        # For each result, create a dictionary with desired fields
+        for item in results[:limit]:
+            result = {field: getattr(item, field, None) for field in search_fields}
```

Comment:
We should make the results an explicit type (see examples [here](https://github.com/refstudio/refstudio/blob/main/python/sidecar/typing.py)). Maybe a `S2SearchResult` type? What do you think @shauryr @cguedes

Reply:
I think that the response type should be agnostic of being a scholar paper search. In addition we should maybe have a field for the search provider (now Scholar) and trying to have the output field names not using s2 specific fields (ex: openAccessPdf -> pdfUrl`

I've noticed that the `paperId` is the primary id, and that `corpusId` is a second way to identify papers. Should we keep both or one of them (`paperId` imo)

ex of using the paperId to get paper details: https://api.semanticscholar.org/api-docs/#tag/Paper-Data/operation/get_graph_get_paper

Reply:
Great suggestion! we can just keep one id, `paperId` @cguedes 
`paperId` will also be the same as the SHA1 of a pdf file (sometimes)